### ISSUE TITLE
map "go to homepage"

### ISSUE THREAD

**Issue by mimooh** [philc/vimium#2528]
How would I create a mapping
gh   - "go to homepage"? 

Firefox vim plugins had this useful command.

**Comment by smblott-github** [comment]
`goToRoot` might be what you're looking for.  To see *all* available commands:

- Go to the options page.
- Click *Show available commands*.
- And make sure *Show advanced commands* is selected.

**Comment by mimooh** [comment]
You showed me where to setup mappings and thanks for that! But what your
proposed option, goToRoot does, is it cuts the path after the site name
and transforms:

	http://www.site.com/a/b/c

to

	http://www.site.com/

So it doesn't resolve my issue. I want to go to where chrome's alt+home
(or clicking on home icon) redirects me - my homepage.

-- 
Karol Kreński

**Comment by smblott-github** [comment]
... in that case, a bookmark (and the Vomnibar) is the only way currently.

**Comment by bad63r** [comment]
Is this still not possible to do as @mimooh wanted?

**Comment by introom** [comment]
instead of limiting it to the homepage, the command should be something like
gotoPage https://foo.bar

**Comment by mrmr1993** [comment]
`map gh createTab https://your.homepage/`

See #2318 for the PR.

**Comment by introom** [comment]
@mrmr1993 createTab creates a new tab.  instead, the desired behavior should be open on the current tab.   one could create a tab and close the current, but that is a little awkward.

**Comment by ljohnston** [comment]
I've gotta agree with @introom ... the following is nice: 

`map gh createTab https://your.homepage/`

But it would also be nice to open urls in the current tab.

**Comment by smblott-github** [comment]
> But it would also be nice to open urls in the current tab.

I agree that this feature would be useful, but it doesn't seem to warrant a whole new command.

What about..?

    map gh createTable URL current

**Comment by wviechtb** [comment]
This would be a really nice feature. Right now, opening a particular URL in the current tab requires several keystrokes (e.g., open Vomnibar, one or more keys to find the desired URL, enter; or open URL in new tab, close current tab, switch to new tab). I have a handful or URLs that I visit regularly and mapping each of them to a single key would be very useful.

**Comment by gdh1995** [comment]
@wviechtb  now `map gh createTab http://www.qq.com` works well.

**Comment by wviechtb** [comment]
@gdh1995 It creates a new tab. As discussed above, the goal is to have a key that goes to a particular URL on the current tab.

**Comment by mimooh** [comment]
This is how it worked out for me, first the issues how I see them:

1. Right after chrome is launched vimium is sadly disabled for security reasons. Second and next pages is where vimium is getting enabled. 

2. When starting a new blank tab vimium is disabled again on this tab.

Both issues are resolved by installing 'new tab url' extension https://chrome.google.com/webstore/detail/new-tab-url/njigpponciklokfkoddampoienefegcl

This extensions seems to make a redirect to a user-provided URL. The simple act of the redirect makes it a second page so vimium gets loaded.

I end up with having my preferred homepage (cannot be file://, must http(s)://) and I have chrome configured to always show my preferred homepage on startup and each newtab. Vimium works everywhere then.

**Comment by wviechtb** [comment]
@mimooh What does this have to do with the suggestion to add a new command (or as @smblott-github suggested, an argument like `current` for the `createTab` command) to go to a user-specified URL in the current tab? Vimium works perfectly fine for me when I launch my browser or open a new tab without having to install another extension and I can set my preferred homepage to anything I like. But this isn't the issue here.

**Comment by mimooh** [comment]
OK, maybe that's too much offtopic, it's how I approach "go to homepage".

**Comment by gdh1995** [comment]
I'm writing [Vimium C](https://github.com/gdh1995/vimium-c), and it supports "map ... openUrl url=http://g.com reuse=0" to open url in current tab. It also overrides browser's new tab URL like the "New Tab URL" extension mentioned above, to make Vimium, and you can configure the focus point of a new tab: focus the address bar / focus the page.

**Comment by mimooh** [comment]
Oh, Vimium C is nice with this reuse=0 option!

**Comment by nsklaus** [comment]
i'm also interested in that issue. please add a way to map a key to load desired url (including local file://) in current tab.

**Comment by captnblood** [comment]
has this issue been resolved ?
is it a wontfix, or is there a way of having a gohome key in vimium?

**Comment by WGUNDERWOOD** [comment]
To open my homepage in the current tab in Chrome with Vimium, I simply have the url saved as a bookmark labelled `b`, and then go to it with `b b RET`.

**Comment by sk29110** [comment]
Is it possible to specify the relative path for this? I do manage a lot of wordpress site and go to plugin pages often so when I type in plugin I redirect to http://currentdomain/plugins.
